### PR TITLE
[DOC-125] added diagnostics page for Voyager

### PR DESCRIPTION
**Pull request by aishwarya24**
@netlify /preview/yugabyte-voyager/diagnostics-report/

### CODE REVIEW

File: docs/content/preview/yugabyte-voyager/install-yb-voyager.md [hunk 1]
```diff
@@ -107,6 +107,10 @@ Install yb-voyager on a machine which satisfies the [Prerequisites](#prerequisit
   </div>
 </div>
 
+### Collect diagnostics
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-voyager/install-yb-voyager.md:110]: Does diagnostics deserve its own top level page, or should all the info just go here

File: docs/content/preview/yugabyte-voyager/install-yb-voyager.md [hunk 2]
```diff
@@ -107,6 +107,10 @@ Install yb-voyager on a machine which satisfies the [Prerequisites](#prerequisit
...
+
+yb-voyager by default captures a [Diagnostics report](../diagnostics-report/) using YugabyteDB diagnostics service that runs each time you use the yb-voyager command. If you don't want to send diagnostics when you run yb-voyager, set the [--send-diagnostics flag](../diagnostics-report/#configuration-flag) to false.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-voyager/install-yb-voyager.md:112]:
```suggestion
By default, yb-voyager captures a [diagnostics report](../diagnostics-report/) using the YugabyteDB diagnostics service that runs each time you use the yb-voyager command. If you don't want to send diagnostics when you run yb-voyager, set the [--send-diagnostics flag](../diagnostics-report/#configuration-flag) to false.
```

File: docs/content/preview/yugabyte-voyager/migrate/bulk-data-load.md [hunk 3]
```diff
@@ -38,15 +38,40 @@ yb-voyager import data file --export-dir <EXPORT_DIR> \
        --escape-char <ESCAPE_CHAR> \ # for csv format only. default: '"'
        --quote-char <QUOTE_CHAR> \ # for csv format only. default: '"'
        --has-header \ # for csv format only. default: false
-       --null-string "<NULL_STRING>"
+      --null-string <NULL_STRING> # default '' (empty string) for csv and '\N'  for text
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-voyager/migrate/bulk-data-load.md:41]: indentation

File: docs/content/preview/yugabyte-voyager/migrate/bulk-data-load.md [hunk 4]
```diff
@@ -38,15 +38,40 @@ yb-voyager import data file --export-dir <EXPORT_DIR> \
...
 ```
 
 Refer to [import data file](../../reference/bulk-data-load/import-data-file/) for details about the arguments.
 
+### Load multiple files into the same table
+
+The import data file command also supports importing multiple files to the same table by providing the `--file-table-map` flag `<fileName>:<tableName>` entry for each file, or by passing a glob expression in place of the file name. For example, `fileName1:tableName,fileName2:tableName` or `fileName*:tableName`.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-voyager/migrate/bulk-data-load.md:48]:
```suggestion
The import data file command also supports importing multiple files to the same table by providing the `--file-table-map` flag with a `<fileName>:<tableName>` entry for each file, or by passing a glob expression in place of the file name. For example, `fileName1:tableName,fileName2:tableName` or `fileName*:tableName`.
```

File: docs/content/preview/yugabyte-voyager/migrate/bulk-data-load.md [hunk 5]
```diff
@@ -38,15 +38,40 @@ yb-voyager import data file --export-dir <EXPORT_DIR> \
...
+
 ### Incremental data loading
 
-The `import data file` command also supports importing multiple files to the same table by providing the --file-table-map flag `<fileName>:<tableName>` entry for each file, or by passing a glob expression in place of the file name.
-For example, `fileName1:tableName,fileName2:tableName` OR `fileName*:tableName`.
+The import data file command also supports importing files to the same table across multiple runs. For example,
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-voyager/migrate/bulk-data-load.md:52]:
```suggestion
You can also import files to the same table across multiple runs. For example, you could import `orders1.csv` as follows:
```

File: docs/content/preview/yugabyte-voyager/migrate/bulk-data-load.md [hunk 6]
```diff
@@ -38,15 +38,40 @@ yb-voyager import data file --export-dir <EXPORT_DIR> \
...
+
+```sh
+yb-voyager import data file --file-table-map 'orders1.csv:orders' ...
+```
+
+followed by,
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-voyager/migrate/bulk-data-load.md:58]:
```suggestion
And then subsequently import `orders2.csv` to the same table as follows:
```

File: docs/content/preview/yugabyte-voyager/migrate/bulk-data-load.md [hunk 7]
```diff
@@ -38,15 +38,40 @@ yb-voyager import data file --export-dir <EXPORT_DIR> \
...
+
+```sh
+yb-voyager import data file --file-table-map 'orders2.csv:orders' ...
+```
+
+If you want to import an updated version of the same file (that is, having the same file name and data-dir), use the `--start-clean` flag and proceed without truncating the table. `yb-voyager` will ingest the data present in the file in upsert mode. For example,
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-voyager/migrate/bulk-data-load.md:64]:
```suggestion
To import an updated version of the same file (that is, having the same file name and data-dir), use the `--start-clean` flag and proceed without truncating the table. yb-voyager ingests the data present in the file in upsert mode. For example:
```

File: docs/content/preview/yugabyte-voyager/overview.md [hunk 8]
```diff
@@ -60,14 +60,6 @@ The following table describes the supported target databases based on your migra
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-voyager/overview.md:60]: The following table shows the target database support for each migration type.

File: docs/content/preview/yugabyte-voyager/reference/cutover-archive/archive-changes.md [hunk 9]
```diff
@@ -30,6 +30,6 @@ The valid *arguments* for archive changes are described in the following table:
 | --fs-utilization-threshold <percentage> | Disk utilization threshold in percentage. (default: 70) |
 | -h, --help | Command line help for archive changes. |
 | --move-to <path> | Destination path to move exported data to. |
-| --send-diagnostics | Send diagnostics information to Yugabyte. (default: true) |
+| --send-diagnostics | Send [diagnostics](../../../diagnostics-report/) information to Yugabyte. (default: true) |
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-voyager/reference/cutover-archive/archive-changes.md:33]: cutover commands don't have this flag?

Comment (aishwarya24) [reply docs/content/preview/yugabyte-voyager/reference/cutover-archive/archive-changes.md:33]: cutover commands don't have it as of version 1.5. @priyanshi-yb any plans of including it later ?

File: docs/content/preview/yugabyte-voyager/diagnostics-report.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+description: Enable diagnostics to collect migration report from the YugabyteDB diagnostics service.
+menu:
+  preview_yugabyte-voyager:
+    identifier: diagnostics-report
+    parent: yugabytedb-voyager
+    weight: 104
+type: docs
+---
+
+YugabyteDB Voyager reports migration diagnostics to the YugabyteDB diagnostics service every time any migration starts through YugabyteDB Voyager. YugabyteDB diagnostics service runs when yb-voyager commands are executed, and stores the data yb-voyager captures with the [--send-diagnostics flag](#configuration-flag).
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/preview/yugabyte-voyager/diagnostics-report.md:13]: @priyanshi-yb - Can you elaborate exactly what the service does. What happens each time a voyager command is run with the flag . What details get transmitted to YB?

[the remark this answers is not in the document]
Comment (priyanshi-yb) [reply docs/content/preview/yugabyte-voyager/diagnostics-report.md]: YB diagnostics service is basically all time running cluster that stores the information collected during the migration. When each command is run with this flag set, the data corresponding to that command is collected and stored on the cluster. ex- `export` command stores the information related to source db.

File: docs/content/preview/yugabyte-voyager/reference/cutover-archive/archive-changes.md (hunk 9 above)
Comment (priyanshi-yb) [reply docs/content/preview/yugabyte-voyager/reference/cutover-archive/archive-changes.md:33]: No, not yet. will add this in the future if required.

File: docs/content/preview/yugabyte-voyager/diagnostics-report.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (ddhodge) [reply docs/content/preview/yugabyte-voyager/diagnostics-report.md]: So it isn't a call home feature? The diagnostics info stays local to their DB?

[the remark this answers is not in the document]
Comment (priyanshi-yb) [reply docs/content/preview/yugabyte-voyager/diagnostics-report.md]: no no, it's a call home feature. It is not stored in their DB. It is stored in the call home cluster we have.

File: docs/content/preview/yugabyte-voyager/diagnostics-report.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    "total_rows": 11927,
+    "total_size": 225422,
+    "largest_table_rows": 11907,
+    "largest_table_size": 224568,
+}
+```
+
+## Configure diagnostics collection
+
+To control whether to send diagnostics to YugabyteDB, you can use the `--send-diagnostics` flag or export an environment variable.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-voyager/diagnostics-report.md:88]:
```suggestion
To control whether to send diagnostics to Yugabyte, you can use the `--send-diagnostics` flag or export an environment variable.
```

**Pull request merged**